reports-rest/src/main/java/org/jboss/da/rest/reports/Reports.java

```diff
@@ -61,15 +65,31 @@
     @Inject
     private BlackArtifactService blackArtifactService;
 
+    @Inject
+    private PomAnalyzer pomAnalyzer;
+
     @POST
     @Path("/scm")
     @Consumes(MediaType.APPLICATION_JSON)
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = "Get dependency report for a project specified in a repository URL",
-            response = Report.class, hidden = true)
-    // TODO unhide when the method will be implemented
-    public Report scmGenerator(SCMRequest scmRequest) {
-        throw new NotImplementedException();
+            response = ScmGAVRelationship.class, hidden = false)
+    public Response scmGenerator(@ApiParam(value = "test") SCMLocator scm)
+            throws Exception {
+
+        try {
+
+            Optional<List<ScmGAVRelationship>> artifactReport = reportsGenerator.getReportFromSCM(scm);
+
+            return artifactReport
+                    .map(x -> Response.ok().entity(artifactReport.get()).build())
+                    .orElseGet(() -> Response.status(Status.NOT_FOUND)
+                            .entity(new ErrorMessage("No relationship found")).build());
+
+        } catch (ScmException|PomAnalysisException|IllegalArgumentException e) {
+            e.printStackTrace();
```

Comment:
Only just a note not to forget: Change it to log, but I'm sure you would do it anyway ;-)